tests/Unit/Artwork/Modules/Project/Services/ProjectServiceTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Tests\Unit\Artwork\Modules\Project\Services;
+
+use Artwork\Modules\CostCenter\Models\CostCenter;
+use Artwork\Modules\Project\Models\Project;
+use Artwork\Modules\Project\Services\ProjectService;
+use Artwork\Modules\User\Models\User;
+use Tests\TestCase;
+
+class ProjectServiceTest extends TestCase
```

Comment:
Es handelt sich hier leider nicht um einen Unit-Test.

Ein Unit-Test ist dadurch charakterisiert, dass du Funktionalitäten im detail testest.

Hier hast du einen Feature-Test. Du sorgst für ein gewisses Umfeld und schaust dir dann das Endergebnis an.

Beim Unit-Test gibst du z.B. vor, wie oft eine Funktion innerhalb der zu testenden Unit aufgerufen wird. Hast du ein Request-Objekt auf dem 5x "boolean" aufgerufen wird (wie bei den ProjectSettings), "$request->boolean('x')", dann assertest du im Unit-Test, dass diese Funktion 5x aufgerufen wird und dabei assertest du auch noch die Keys die verwendet werden. Welche Werte die Funktionen zurückgeben steuerst du. Wird also irgendwann ein weiterer Request Parameter hinzugefügt oder entfernt schlägt der Test sofort fehl.

Dazu mockst du das Request Objekt was verwendet wird, somit steuerst du welche Werte in die Service-Funktion kommen. Dann weißt du auch wie die Funktion sich verhalten muss und was sie im exakten tun muss.